packages/dd-trace/src/platform/node/metrics.js

```diff
@@ -255,54 +255,88 @@ function captureCommonMetrics () {
   captureHistograms()
 }
 
+function unpackHistogram (buffer) {
+  return {
+    min: buffer[0],
+    max: buffer[1],
+    sum: buffer[2],
+    avg: buffer[3],
+    count: buffer[4],
+    median: buffer[5],
+    p95: buffer[6]
+  }
+}
+
+// https://source.chromium.org/chromium/chromium/src/+/master:v8/include/v8.h?q=GCType
+const kGCTypeScavenge = 1 << 0
+const kGCTypeMarkSweepCompact = 1 << 1
+const kGCTypeIncrementalMarking = 1 << 2
+const kGCTypeProcessWeakCallbacks = 1 << 3
+const kGCTypeAll = kGCTypeScavenge | kGCTypeMarkSweepCompact | kGCTypeIncrementalMarking | kGCTypeProcessWeakCallbacks
+const gcTypeNames = {
+  [kGCTypeScavenge]: 'scavenge',
+  [kGCTypeMarkSweepCompact]: 'mark_sweep_compact',
+  [kGCTypeIncrementalMarking]: 'incremental_marking',
+  [kGCTypeProcessWeakCallbacks]: 'process_weak_callbacks',
+  [kGCTypeAll]: 'all'
+}
+
 function captureNativeMetrics () {
-  const stats = nativeMetrics.stats()
-  const spaces = stats.heap.spaces
+  const rawStats = nativeMetrics.dump(
+    nativeMetrics.strings,
+    nativeMetrics.processBuffer,
+    nativeMetrics.eventLoopBuffer
+  )
+
   const elapsedTime = process.hrtime(time)
 
   time = process.hrtime()
 
   const elapsedUs = elapsedTime[0] * 1e6 + elapsedTime[1] / 1e3
-  const userPercent = 100 * stats.cpu.user / elapsedUs
-  const systemPercent = 100 * stats.cpu.system / elapsedUs
+  const userPercent = 100 * nativeMetrics.processBuffer[0] / elapsedUs
```

Comment:
This is harder to read. Can each buffer chunk be named?

Reply:
I'm not sure what you mean.

Reply:
There is no way when reading `nativeMetrics.processBuffer[0]` to know what it contains. It would be clearer to assign it to a variable with a more representative name. This also applies to the other elements of the array.